### PR TITLE
Support for streaming ollama generator

### PR DESCRIPTION
**Pull request by sachinsachdeva**
Implements #167 

- [x] Support Streaming for ollama generator
- [x] Add/update tests
- [x]  implement serializing/deserializing for OllamaGenerator

### CODE REVIEW

**Comment by anakin87** [comment]
Hey, @sachinsachdeva!
Thanks for working on this...

When you want this PR to be reviewed, put it in "Ready for review" status.

In the meantime, I would say that now that the Generator also accepts a streaming callback, we should also take care of properly serializing/deserializing this component. You can refer to the methods `to_dict`/`from_dict` of the [OpenAI Generator](https://github.com/deepset-ai/haystack/blob/main/haystack/components/generators/openai.py#L110).

Let me know if you need any help or directions.

**Comment by sachinsachdeva** [comment]
@anakin87 ready for review now

File: integrations/ollama/src/haystack_integrations/components/generators/ollama/generator.py [hunk 1]
```diff
@@ -71,13 +109,54 @@ def _convert_to_haystack_response(self, ollama_response: Response) -> Dict[str,
...
         """
+
         resp_dict = ollama_response.json()
 
         replies = [resp_dict["response"]]
         meta = {key: value for key, value in resp_dict.items() if key != "response"}
 
         return {"replies": replies, "meta": [meta]}
 
+    def _convert_to_response(self, chunks: List[StreamingChunk]) -> Dict[str, List[Any]]:
```
[the remark this answers is not in the document]
Comment (sachinsachdeva) [reply integrations/ollama/src/haystack_integrations/components/generators/ollama/generator.py]: done

File: integrations/ollama/tests/test_generator.py [hunk 2]
```diff
@@ -79,20 +146,45 @@ def test_init(self):
...
             "template": configuration["template"],
             "options": {},
         }
 
         assert observed == expected
+
+
+class TestOllamaStreamingGenerator:
+    @pytest.mark.integration
+    def test_ollama_generator_run_streaming(self):
```
[the remark this answers is not in the document]
Comment (sachinsachdeva) [reply integrations/ollama/tests/test_generator.py]: no specific reason (anymore :) ), moved to main test class

**Review by ZanSara — APPROVED** [review]
I tested it and it works well. Thank you for your contribution! :hugs:

**Pull request merged**